With wrapped mariadb_version.h, FindMariaDBClient.cmake fails

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
Cannot compile Kodi on Gentoo with MariaDB because `/usr/include/mariadb/mariadb_version.h` doesn't contain the string `MARIADB_CLIENT_VERSION_STR` (I suspect this impact other distros, but haven't found out for sure)

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
`USE="mariadb" emerge =media-tv/kodi-9999`
FindMariaDBClient.cmake module does not find what it needs. The module searches for `MARIADB_CLIENT_VERSION_STR` in `/usr/include/mariadb/mariadb_version.h`, but that file is just a wrapper header to include `/usr/include/x86_64-pc-linux-gnu/mariadb/mariadb_version.h` where the constant is actually defined. As the cmake module does not "parse" the wrapped header file but reads it line by line and searches for the constant with a regexp, cmake fails in the end.
Here's what `/usr/include/mariadb/mariadb_version.h` looks like: https://pastebin.com/h81cta5x
This header file is wrapped as that's how multilib support is implemented.

## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
You should be able to compile Kodi on Gentoo :)

## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->
Is extracting the version from the header file really necessary? Is there a better way to get the version? Is getting the version necessary at all?

### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1.  Compile Kodi 18 (current master) on a Gentoo system with MariaDB support

### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:

Since this is a build issue, a debug log doesn't appear to be applicable.

### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->

Since this is a build issue, a screenshot doesn't appear to be applicable.

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->



### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberri-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Gentoo Linux ~amd64
 - Kodi version: 18 (master, currently c997272e9edd34c1efb879b5f5ee2936c7a3e528 )



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

I tried https://github.com/xbmc/xbmc/pull/14602 and that didn't fix the problem. The output from that run is:
```
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find MariaDBClient (missing: MARIADBCLIENT_LIBRARY) (found
  version "10.3.9")
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindMariaDBClient.cmake:52 (find_package_handle_standard_args)
  cmake/scripts/common/Macros.cmake:374 (find_package)
  cmake/scripts/common/Macros.cmake:442 (find_package_with_ver)
  CMakeLists.txt:187 (core_optional_dep)
```

In case it helps, here are all the files I have regarding the mariadb client; I can provide contents too, course:
```
/usr/bin/mariadb_config -> x86_64-pc-linux-gnu-mariadb_config
/usr/bin/x86_64-pc-linux-gnu-mariadb_config
/usr/include/mariadb/errmsg.h
/usr/include/mariadb/ma_list.h
/usr/include/mariadb/ma_pvio.h
/usr/include/mariadb/ma_tls.h
/usr/include/mariadb/mariadb/ma_io.h
/usr/include/mariadb/mariadb_com.h
/usr/include/mariadb/mariadb_ctype.h
/usr/include/mariadb/mariadb_dyncol.h
/usr/include/mariadb/mariadb_stmt.h
/usr/include/mariadb/mariadb_version.h
/usr/include/mariadb/mysql.h
/usr/include/mariadb/mysql/client_plugin.h
/usr/include/mariadb/mysql/plugin_auth.h
/usr/include/mariadb/mysql/plugin_auth_common.h
/usr/include/mariadb/mysqld_error.h
/usr/include/x86_64-pc-linux-gnu/mariadb/mariadb_version.h
/usr/lib/debug/usr/bin/x86_64-pc-linux-gnu-mariadb_config.debug
/usr/lib/debug/usr/lib64/libmariadb.so.3.debug
/usr/lib/debug/usr/lib64/mariadb/plugin/dialog.so.debug
/usr/lib/debug/usr/lib64/mariadb/plugin/mysql_clear_password.so.debug
/usr/lib/debug/usr/lib64/mariadb/plugin/remote_io.so.debug
/usr/lib/debug/usr/lib64/mariadb/plugin/sha256_password.so.debug
/usr/lib64/libmariadb.so -> libmariadb.so.3
/usr/lib64/libmariadb.so.3
/usr/lib64/mariadb/plugin/dialog.so
/usr/lib64/mariadb/plugin/mysql_clear_password.so
/usr/lib64/mariadb/plugin/remote_io.so
/usr/lib64/mariadb/plugin/sha256_password.so
```

What further information can I provide to debug this issue?

--- a/cmake/modules/FindMariaDBClient.cmake
+++ b/cmake/modules/FindMariaDBClient.cmake
@@ -21,15 +21,25 @@ else()
   set(EXTRA_FIND_ARGS)
 endif()
 
-find_path(MARIADBCLIENT_INCLUDE_DIR NAMES mariadb/mysql.h mariadb/server/mysql.h)
-find_library(MARIADBCLIENT_LIBRARY_RELEASE NAMES mariadbclient libmariadb
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(PC_MARIADBCLIENT mariadb QUIET)
+endif()
+
+
+find_path(MARIADBCLIENT_INCLUDE_DIR NAMES mariadb/mysql.h mariadb/server/mysql.h
+                                           PATHS ${PC_MARIADBCLIENT_INCLUDEDIR})
+find_library(MARIADBCLIENT_LIBRARY_RELEASE NAMES mariadbclient mariadb
+                                           PATHS ${PC_MARIADBCLIENT_LIBDIR}
                                            PATH_SUFFIXES mariadb
                                            ${EXTRA_FIND_ARGS})
-find_library(MARIADBCLIENT_LIBRARY_DEBUG NAMES mariadbclient libmariadb
+find_library(MARIADBCLIENT_LIBRARY_DEBUG NAMES mariadbclient mariadb
+                                         PATHS ${PC_MARIADBCLIENT_LIBDIR}
                                          PATH_SUFFIXES mariadb
                                          ${EXTRA_FIND_ARGS})
 
-if(MARIADBCLIENT_INCLUDE_DIR AND EXISTS "${MARIADBCLIENT_INCLUDE_DIR}/mariadb/mariadb_version.h")
+if(PC_MARIADBCLIENT_VERSION)
+  set(MARIADBCLIENT_VERSION_STRING ${PC_MARIADBCLIENT_VERSION})
+elseif(MARIADBCLIENT_INCLUDE_DIR AND EXISTS "${MARIADBCLIENT_INCLUDE_DIR}/mariadb/mariadb_version.h")
   file(STRINGS "${MARIADBCLIENT_INCLUDE_DIR}/mariadb/mariadb_version.h" mariadb_version_str REGEX "^#define[\t ]+MARIADB_CLIENT_VERSION_STR[\t ]+\".*\".*")
   string(REGEX REPLACE "^#define[\t ]+MARIADB_CLIENT_VERSION_STR[\t ]+\"([^\"]+)\".*" "\\1" MARIADBCLIENT_VERSION_STRING "${mariadb_version_str}")
   unset(mariadb_version_str)

